### PR TITLE
Update NixOS Python guidance

### PR DESCRIPTION
**Pull request by tfoote**
Following clarifying comments in https://github.com/ros/rosdistro/pull/38539#issuecomment-1718489729

### CODE REVIEW

**Comment by tfoote** [comment]
@lopsided98 @hacker1024 @kjeremy for awareness/review

**Review by clalancette — APPROVED** [review]
Seems reasonable to me, but I'll wait for further confirmation before merging.

**Comment by kjeremy** [comment]
Looks good to me.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -200,6 +200,7 @@ Work has been proposed to add a separate installer for AUR packages [ros-infrast
 
 * [NixOS unstable channel](https://github.com/NixOS/nixpkgs/tree/nixos-unstable), search available at https://search.nixos.org/packages
 * [nix-ros-overlay](https://github.com/lopsided98/nix-ros-overlay)
+* Following the [NixOS Python Guide](https://nixos.org/manual/nixpkgs/stable/#python) Use `pythonPackages` for Python 2 and `python3Packages` for Python 3 keys.
```
Comment (hacker1024) [inline CONTRIBUTING.md:203]:
```suggestion
* Following the [NixOS Python Guide](https://nixos.org/manual/nixpkgs/stable/#python), use `pythonPackages` for Python 2 and `python3Packages` for Python 3 keys.
```

Erroneous capitalization?

**Pull request merged**